sigmastate/src/main/scala/org/ergoplatform/ErgoBoxCandidate.scala

```diff
@@ -72,35 +87,49 @@ class ErgoBoxCandidate(val value: Long,
     }
   }
 
-  override def hashCode(): Int =
+  // TODO HF: fix hashCode, it should be consisten with [[equals]] and use [[bytesWithNoRef]]
```

Comment:
Unlikely this is HF-related issue

Reply:
Can you claim it is "definitely not HF-related"? If yes, I can remove HF prefix, but I'm not so sure.

Reply:
@aslesarenko This is you job to explain why this is HF-related, as you made the claim

Reply:
I don't want to spend time on this analysis now as part of this PR. It is out of it's scope and goal. 
I found this place and marked it HF in order to come to it later as part of HF enabling work.
If you have simple arguments why HF can be removed, I will be happy to do that, otherwise, let's leave it as is.

Reply:
@aslesarenko you've basically put a big red flag with the comment. If transaction validation can depend on standard hash function collision-resistance, the protocol has big problems. Please do such statements for a reason.

Reply:
I don't understand, how this relates to collision resistance?

Reply:
changes to TODO refactor with explaination

Reply:
@aslesarenko that's the basic property of hash functions, both standard and cryptographic. Requirements are different though. Anyway. the protocol MUST not be aware about standard HF you're using for doing data structures efficiently in the clients and its CR.